server/src-lib/Hasura/RQL/DML/Select/Internal.hs

```diff
@@ -153,6 +153,25 @@ buildJsonObject pfx parAls arrRelCtx flds =
                      ANIField (fldAls, arrSel)
         in S.mkQIdenExp arrPfx fldAls
 
+    toSQLCol :: PGColInfo -> S.SQLExp
+    toSQLCol col = toJSONableExp (pgiType col) $
+                    S.mkQIdenExp (mkBaseTableAls pfx) $ pgiName col
+
+    toSQLColArgs :: PGColInfo -> AnnArgs -> S.SQLExp
+    toSQLColArgs col args
+      | isJSONType colTy || isJSONBType colTy =
```

Comment:
You don't need to check `isJSONType` or `isJSONBType` since it is guaranteed that`PGColArg` is applied to only `jsonb` and `json` column types from `GraphQL` schema generation.

Reply:
I got it. However, would we leave that condition for future support for custom argument types? IMO

Reply:
We're generating `GraphQL` schema based on column types. However, if we use `ColOp` kind of data type I suggested, then we don't require any column type check. And, `toSQLCol` and `toSQLColArgs` functions will get merged. In this case, the `operation` type and the RHS `SQLExp` are decided at the time of resolving the query in `Resolve/Select.hs` module. Currently we're generating only one operator ie. `path` for `json` and `jsonb` types.